### PR TITLE
Fixes for Transitions + Sidebar hover effect speedup.

### LINKED ISSUE #2485: Multiple transitions running for a AvaloniaProperty change
I have an AvaloniaProperty for which I have created a transition that runs for 500ms with QuarticEaseOut.
The property changes whenever the user interacts with a Canvas by dragging or zooming with mouse or touch.

Every time the property changes a new TransitionInstance is created running in parallel with any previous unfinished TransitionInstances.

This leads to the property value jumping around as the multiple TransitionInstances fight over the current value.

Please could we have an option to cancel the running TransitionInstance when creating a new TransitionInstance for a property change?

### LINKED ISSUE #2490: Transitions don't always transition to their final value
I have noticed that when a transition is applied to a AvaloniaProperty and the property value is changed, the final value is not always reached when the transition completes.

For example: I change a double property from 0 to 10 with a transition set to QuarticEasing and duration 1s, the final value of the property may sometimes be less than 10 when the transition completes.
The problem appears to be in the TransitionInstance.TimerTick method as it may never be called with t.Ticks == _duration.Ticks.

I have added PublishNext(1.0) to the method in my local copy and this seems to solve the problem for me:
 
```
private void TimerTick(TimeSpan t)
{
    var interpVal = (double)t.Ticks / _duration.Ticks;

    if (interpVal > 1d || interpVal < 0d)
    {
        PublishNext(1.0);   // Set the final property value to the finish value
        PublishCompleted();
        return;
    }

    PublishNext(interpVal);
}
```

### PR DESCRIPTION
**Pull request by jmacato**
## What does the pull request do?

Addresses issues with multiple instances of `TransitionInstance` not being disposed and also the issue where transitions doesn't finish cleanly. Also speed-up the sidebar hover transitions since it seems to be quite slow.

## What is the current behavior?

- Multiple  `TransitionInstance` objects are retained even when the transition itself is finished.  
- Transitions sometimes overshoots and doesnt publishes the final interpolation value when the transition is finished.

## What is the updated/expected behavior with this PR?

- `TransitionInstance`'s dispose method should be called correctly when finished.
- Transition's interpolation values are clamped.

## Fixed issues

Fixes #2485 
Fixes #2490

### CODE REVIEW

File: src/Avalonia.Animation/Animatable.cs [hunk 1]
```diff
@@ -36,6 +36,9 @@ public IClock Clock
 
         private Transitions _transitions;
 
+        private Dictionary<AvaloniaProperty, IDisposable> _previousTransitions
```
Comment (grokys) [inline src/Avalonia.Animation/Animatable.cs]: Perhaps don't allocate this collection until it's needed? Many controls won't have transitions and it'll save quite a bit of memory.

Comment (grokys) [reply src/Avalonia.Animation/Animatable.cs]: Also will this prevent transitions from being GC'd where previously they would have been?

**Review by grokys — CHANGES_REQUESTED** [review]

File: src/Avalonia.Animation/Animatable.cs (hunk 1 above)
Comment (jmacato) [reply src/Avalonia.Animation/Animatable.cs]: Yep, i forgot to include the local diff for addressing that

File: tests/Avalonia.Animation.UnitTests/TransitionsTest.cs [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+using Avalonia.Controls;
+using Avalonia.Styling;
+using Avalonia.UnitTests;
+using Avalonia.Data;
+using Xunit;
+using Avalonia.Animation.Easings;
+
+namespace Avalonia.Animation.UnitTests
+{
+    public class TransitionsTest
```
Comment (grokys) [inline tests/Avalonia.Animation.UnitTests/TransitionsTest.cs]: Nit: all other unit test classes are named with the plural, so `TransitionsTests`.

**Review by grokys — CHANGES_REQUESTED** [review]
Just a tiiiiiiny nit and we're good to go!

**Review by grokys — APPROVED** [review]

**Pull request merged**